src/components/KInput/KInput.vue

```diff
@@ -35,18 +35,37 @@
         v-bind-once="{ id: inputId, ...(helpText && { 'aria-describedby': helpTextId }) }"
         :aria-invalid="error || hasError || charLimitExceeded ? 'true' : undefined"
         class="input"
-        :type="type"
+        :type="inputType"
         v-bind="modifiedAttrs"
         :value="getValue()"
         @input="handleInput"
       >
 
       <div
-        v-if="$slots.after"
+        v-if="$slots.after || (type === 'password' && showPasswordMaskToggle)"
         ref="afterSlotElement"
         class="after-content-wrapper"
       >
-        <slot name="after" />
+        <button
+          v-if="type === 'password' && showPasswordMaskToggle"
+          :aria-label="`${maskValue ? 'Hide' : 'Show'} value`"
+          class="mask-value-toggle-button"
+          @click.stop="maskValue = !maskValue"
+        >
+          <VisibilityIcon
+            v-if="maskValue"
+            decorative
+          />
+          <VisibilityOffIcon
+            v-else
+            decorative
+          />
```

Comment:
Another question: should the icon reflect the current status or action? Current implementation is opposite to our login page's password input:

<img width="387" alt="image" src="https://github.com/user-attachments/assets/5e7f7938-34b4-4c0a-904f-2b918fd2f691">

Reply:
Oh good catch -- I looked at some other website to look up how they handle that but aligning with our login page makes more sense. Fixed.